### PR TITLE
feat: Add date filter to the contributor calendar

### LINKED ISSUE #663: feature: Add date filter to contributor calendar
### Type of feature

Select the type of feature request, the lowercase should also be the PR prefix.

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

### Current behavior
205411267-3c589f8f-9316-4df8-9b55-cefae8637a24

### Suggested solution

Add the same component and logic for contribution data filtering data as the repository page. 

Screen Shot 2022-12-15 at 6 20 51 AM

Describe the solution you'd like.

### Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](https://docs.opensauced.pizza/contributing/code-of-conduct/).

- [x] I agree to follow this project's Code of Conduct

### Contributing Docs

If you plan on contributing code please read - [Contribution Guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/)

- [x]  I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by shamimbinnur**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR:
- Adds the functionality to filter contributor calendar by "7 days", "30 days" and "3 months"

How I did:
- I have used the same date-filtering component used for the repository.
- Added an extra param "range" to the API endpoint.
- Added an extra state "contributorRange" in the store.

## Related Tickets & Documents
Fixes #663 

## Mobile & Desktop Screenshots/Recordings
Before:
Screenshot 2022-12-16 at 11 19 41 PM

After:
Screenshot 2022-12-16 at 11 17 34 PM

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: lib/hooks/useContributionsList.ts [hunk 1]
```diff
@@ -1,16 +1,21 @@
 import useSWR from "swr";
+import useStore from "lib/store";
 
 interface PaginatedContributorsResponse {
   readonly data: DbContribution[];
   readonly meta: Meta;
 }
 
 const useContributionsList = (repoId: string, limit = "", orderBy = "") => {
+  const range = useStore(state => state.contributorRange);
```
Comment (brandonroberts) [inline lib/hooks/useContributionsList.ts:10]: We want to keep the same range across the different tabs of the dashboard, so we can reuse the `range` property and have the other hooks revalidate when it has changed.

File: lib/store/index.ts [hunk 2]
```diff
@@ -19,7 +21,8 @@ const store = create<AppStore>()((set) => ({
   onboardUser: () => set((state) => ({ ...state, onboarded: true })),
   setSession: ({ onboarded, waitlisted }: { onboarded: boolean; waitlisted: boolean }) =>
     set((state) => ({ ...state, onboarded, waitlisted })),
-  updateRange: (range: number) => set((state) => ({ ...state, range }))
+  updateRange: (range: number) => set((state) => ({ ...state, range })),
+  updateContributorRange: (contributorRange: number) => set((state) => ({ ...state, contributorRange }))
```
Comment (brandonroberts) [inline lib/store/index.ts:25]: We can keep this method so we can track the origin of the change.

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by brandonroberts** [comment]
Thanks @shamimbinnur! Left some feedback

**Comment by bdougie** [comment]
@shamimbinnur! Thanks a bunch for jumping in. 

Be sure to follow the [Bugs](https://github.com/orgs/open-sauced/projects/6) or [Papercuts](https://github.com/open-sauced/insights/issues?q=is%3Aopen+is%3Aissue+milestone%3APapercuts).  I plan to add little features like this over there.

**Comment by shamimbinnur** [comment]
> Thanks @shamimbinnur! Left some feedback

Updated and kept the same range for both of the components. Thanks, @brandonroberts

**Comment by shamimbinnur** [comment]
> 

Sure thing! Thanks, @bdougie

File: interfaces/global-state-types.ts [hunk 3]
```diff
@@ -4,5 +4,6 @@ export interface GlobalStateInterface {
   user?: User | null;
   onboarded?: boolean;
   waitlisted?: boolean;
-  range?: number
+  range?: number;
+  contributorRange?: number;
```
Comment (brandonroberts) [inline interfaces/global-state-types.ts:8]:
```suggestion
```

**Review by brandonroberts — APPROVED** [review]
small change

**Comment by shamimbinnur** [comment]
> small change

Forgot to save the change :(
Just updated. @brandonroberts

**Review by bdougie — APPROVED** [review]
UI works. But mentioning that caching would be extremely useful when filtering contributions.

**Comment by brandonroberts** [comment]
SWR is already handling the caching as were are using its hooks for data fetching. More advanced caching in the backend would be useful though.

**Comment by brandonroberts** [comment]
Thanks again @shamimbinnur!

**Pull request merged**